lib/auth.js

```diff
@@ -231,6 +234,47 @@ module.exports = new Model({
     }.bind(this));
   },
 
+  signInWithOAuthRedirect: function (args, window) {
+    // Check to see if we're running in the browser or not
+    if (typeof window === 'undefined') {
+      throw new Error('OAuth login is only available via the browser.')
+    } else if (!args.appId) {
+      throw new Error('You need to specify an app ID.');
+    } else if (!args.redirectUri) {
+      throw new Error('You need to specify a URI to redirect to after authorization.');
+    }
+
+    var originalArguments = arguments;
+    return this.checkCurrent().then(function(user) {
+      if (user) {
+        return this.signOut().then(function() {
+          return this.signIn.apply(this, originalArguments);
+        }.bind(this));
+      } else {
+        console.log('Signing in with OAuth (redirect)');
+        var url = [
+          config.host,
+          '/oauth/authorize',
+          '?response_type=token',
+          '&client_id=',
+          args.appId,
+          '&redirect_uri=',
+          args.redirectUri
+        ].join('');
+        window.location.href = url;
```

Comment:
If we use `location.assign(url)` here we won't have to rebuild and compare strings in the test.